File: test/python/RobotRaconteurTest/ServiceTestClient3.py
from __future__ import absolute_import

import RobotRaconteur as RR
RRN = RR.RobotRaconteurNode.s
import pytest


class ServiceTestClient3:
    def __init__(self):
        pass

    def RunFullTest(self, url):
        self.ConnectService(url)

        self.TestProperties()

        self.TestFunctions()

        self.DisconnectService()

    def ConnectService(self, url):
        self._r = RRN.ConnectService(url)

    def DisconnectService(self):
        RRN.DisconnectService(self._r)

    def TestProperties(self):
        _ = self._r.d1
        self._r.d1 = 3.0819

        try:
            _ = self._r.err
        except RR.InvalidArgumentException as e:
            assert e.message == "Test message 1"
        else:
            assert False

        with pytest.raises(RR.InvalidOperationException):
            self._r.err = 100

    def TestFunctions(self):

        self._r.f1()
        self._r.f2(247)

        res = self._r.f3(1, 2)
        assert res == 3

        with pytest.raises(RR.InvalidOperationException):
            self._r.err_func()

        asynctestexp = RRN.GetExceptionType(
            "com.robotraconteur.testing.TestService5.asynctestexp", self._r)
        try:
            self._r.err_func2()
        except asynctestexp:
            pass
        else:
            assert False

File: test/net/RobotRaconteurNETTest/ServiceTestClient3.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using com.robotraconteur.testing.TestService5;
using RobotRaconteur;
using System.Threading;
using System.IO;
using System.Threading.Tasks;

namespace RobotRaconteurNETTest
{
public class ServiceTestClient3
{

    public void ConnectService(string url)
    {
        r = (asynctestroot)RobotRaconteurNode.s.ConnectService(url);
    }

    public void DisconnectService()
    {
        RobotRaconteurNode.s.DisconnectService(r);
    }

    asynctestroot r;

    public void RunFullTest(string url)
    {
        ConnectService(url);

        TestProperties();

        // TestFunctions();

        DisconnectService();
    }

    public void TestProperties()
    {
        RRAssert.AreEqual(r.d1, 8.5515);

        r.d1 = 3.0819;

        bool thrown1 = false;
        try
        {
            var res = r.err;
        }
        catch (ArgumentException err)
        {
            thrown1 = true;
            RRAssert.AreEqual(err.Message, "Test message 1");
        }
        RRAssert.IsTrue(thrown1);

        bool thrown2 = false;
        try
        {
            r.err = 10;
        }
        catch (InvalidOperationException err)
        {
            thrown2 = true;
            RRAssert.AreEqual(err.Message, "");
        }
        RRAssert.IsTrue(thrown2);
    }

    public void TestFunctions()
    {
        r.f1();
        r.f2(247);

        bool thrown1 = false;
        try
        {
            r.err_func();
        }
        catch (InvalidOperationException)
        {
            thrown1 = true;
        }
        RRAssert.IsTrue(thrown1);

        bool thrown2 = false;
        try
        {
            r.err_func2();
        }
        catch (com.robotraconteur.testing.TestService5.asynctestexp)
        {
            thrown2 = true;
        }
        RRAssert.IsTrue(thrown2);
    }
}
}